src/scss/_main.scss

```diff
@@ -352,31 +373,37 @@
 	@include oGridColspan($span);
 }
 
+@mixin oGridRowMargins($total-cols: $o-grid-columns) {
+	$subgrid-ratio: $o-grid-columns / $total-cols;
+
+	margin-right: auto;
+	margin-left: auto;
+}
+
 /// Base row styles
 ///
 /// @param {String} $grid-mode [$o-grid-mode]
-@mixin oGridRow($grid-mode: $o-grid-mode) {
+@mixin oGridRow($grid-mode: $o-grid-mode, $total-cols: $o-grid-columns) {
 	clear: both;
 	min-width: $o-grid-min-width;
 	// Older browsers get a fixed-width layout
 	max-width: oGridGetMaxWidthForLayout($o-grid-fixed-layout);
-	padding-left: $o-grid-gutter / 2;
-	padding-right: $o-grid-gutter / 2;
 	box-sizing: border-box;
-	margin-left: auto;
-	margin-right: auto;
+
+	@include oGridRowMargins($total-cols);
 
 	// Clearfix
 	zoom: 1;
+	overflow: auto;
```

Comment:
I'm probably not correct on terminology, but it forces the row element grow to contain the children which was the purpose.

I wasn't quite sure if the flexbox branch is just a proposal PR or if that's a definite future of o-grid. Can try to rebase to that if you want